src/components/ProjectSettingsTabs/CreateProjectModal.tsx

```diff
@@ -0,0 +1,179 @@
+import React, { useState } from "react";
+import { useMutation } from "@apollo/client";
+import styled from "@emotion/styled";
+import { Size, Variant } from "@leafygreen-ui/button";
+import { Button } from "components/Button";
+import { ConfirmationModal } from "components/ConfirmationModal";
+import { SpruceForm, SpruceFormProps } from "components/SpruceForm";
+import { size } from "constants/tokens";
+import { useToastContext } from "context/toast";
+import {
+  CreateProjectMutation,
+  CreateProjectMutationVariables,
+} from "gql/generated/types";
+import { CREATE_PROJECT } from "gql/mutations";
+
+const getModalFormDefinition = (owner: string, repo: string) => ({
+  schema: {
+    type: "object" as "object",
+    properties: {
+      projectName: {
+        type: "string" as "string",
+        title: "Project Name",
+      },
+      projectId: {
+        type: "string" as "string",
+        title: "Project Id",
+        value: "hi",
+      },
+      owner: {
+        type: "string" as "string",
+        title: "Owner",
+        default: owner,
+      },
+      repo: {
+        type: "string" as "string",
+        title: "Repo",
+        default: repo,
+      },
+    },
+  },
+  uiSchema: {
+    projectName: {
+      "ui:data-cy": "project-name-input",
+      "ui:showErrors": false,
+    },
+    projectId: {
```

Comment:
This seems like a good opportunity to use LeafyGreen's `optional` prop. There's no support for it yet in SpruceForm's LeafyGreenWidgets file, but it would be great to add!

Reply:
What would be the benefit of using `optional`? This already functions as optional given how the validate function is set up.

Reply:
The input is labeled as optional, example [here](https://www.mongodb.design/component/text-input/example/)!
<img width="371" alt="image" src="https://user-images.githubusercontent.com/9298431/156636644-d2bd258b-25f5-4300-b7a7-0d5ab350b1e3.png">

Reply:
sweet! added.